roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,174 @@
+        namespace: "{{ keycloak_helm_release_namespace }}"
+      type: Opaque
+
+- name: Grab Keycloak database secret
+  kubernetes.core.k8s_info:
+    api_version: v1
+    kind: Secret
+    name: keycloak-db-credentials
+    namespace: "{{ keycloak_helm_release_namespace }}"
+  register: _keycloak_db_secret
+
+- name: Create a Keycloak user
+  community.mysql.mysql_user:
+    login_host: "{{ _pxc_service.resources[0].spec.clusterIP }}"
+    login_user: root
+    login_password: "{{ openstack_helm_endpoints.oslo_db.auth.admin.password }}"
+    name: "{{ _keycloak_db_secret.resources[0]['data']['username'] | b64decode }}"
+    password: "{{ _keycloak_db_secret.resources[0]['data']['password'] | b64decode }}"
+    host: "%"
+    priv: "{{ keycloak_database_name }}.*:ALL"
+
+- name: Disable pxc strict mode
+  community.mysql.mysql_query:
+    login_host: "{{ _pxc_service.resources[0].spec.clusterIP }}"
+    login_user: root
+    login_password: "{{ openstack_helm_endpoints.oslo_db.auth.admin.password }}"
+    query: "set global pxc_strict_mode='PERMISSIVE'"
+
+- name: Deploy Helm chart
+  run_once: true
+  kubernetes.core.helm:
+    name: "{{ keycloak_helm_release_name }}"
+    chart_ref: "{{ keycloak_helm_chart_ref }}"
+    release_namespace: "{{ keycloak_helm_release_namespace }}"
+    create_namespace: true
+    kubeconfig: /etc/kubernetes/admin.conf
+    wait: true
+    values: "{{ _keycloak_helm_values | combine(keycloak_helm_values, recursive=True) }}"
+
+- name: Disable pxc strict mode
+  community.mysql.mysql_query:
+    login_host: "{{ _pxc_service.resources[0].spec.clusterIP }}"
+    login_user: root
+    login_password: "{{ openstack_helm_endpoints.oslo_db.auth.admin.password }}"
+    query: "set global pxc_strict_mode='ENFORCING'"
+
+- name: Create realm and clients
+  when: (_keycloak_realms + keycloak_realms) | length > 0
+  block:
+    - name: Wait until Keycloak ready
+      kubernetes.core.k8s_info:
+        api_version: apps/v1
+        kind: StatefulSet
+        name: keycloak
+        namespace: "{{ keycloak_helm_release_namespace }}"
+      register: _keycloak_statefulest_info
+      until: _keycloak_statefulest_info.resources[0].status.readyReplicas >= 1
+      retries: 60
+      delay: 10
+
+    - name: Create realms
+      # Note(okozachenko1203): Ignoring changes in this task until the upstream issue
+      #                        https://github.com/keycloak/keycloak/issues/20970 is resolved.
+      changed_when: false
+      community.general.keycloak_realm:
+        auth_client_id: admin-cli
+        auth_keycloak_url: "https://{{ keycloak_host }}"
+        auth_realm: master
+        auth_username: "{{ keycloak_admin_username }}"
+        auth_password: "{{ keycloak_admin_password }}"
+        id: "{{ item.id }}"
+        realm: "{{ item.id }}"
+        enabled: true
+      loop: "{{ _keycloak_realms + keycloak_realms }}"
+      retries: 60
+      delay: 5
+      register: _result
+      until: _result is not failed
```

Comment:
i think it would be neat to actually create a role called `keycloak_realm` which would create a realm (similar to the way we `glance_image`) as well, same concept as `keycloak_client` described below..

We can then create another role called `keycloak_atmosphere_realm` which calls `keycloak_realm` with the values needed to create the atmosphere realm, and this would be a dependency for those services that need to use it.